sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -1071,23 +1096,27 @@ def hpat_pandas_series_copy(self, deep=True):
     :obj:`pandas.Series` or :obj:`pandas.DataFrame`
         Object type matches caller.
     """
-    _func_name = 'Method Series.copy().'
 
-    if (isinstance(self, SeriesType) and (isinstance(deep, (types.Omitted, types.Boolean)) or deep)):
-        if isinstance(self.index, types.NoneType):
-            def hpat_pandas_series_copy_impl(self, deep=True):
-                if deep:
-                    return pandas.Series(self._data.copy())
-                else:
-                    return pandas.Series(self._data)
-            return hpat_pandas_series_copy_impl
-        else:
-            def hpat_pandas_series_copy_impl(self, deep=True):
-                if deep:
-                    return pandas.Series(self._data.copy(), index=self._index.copy())
-                else:
-                    return pandas.Series(self._data, index=self._index.copy())
-            return hpat_pandas_series_copy_impl
+    ty_checker = TypeChecker('Method Series.copy().')
+    ty_checker.check(self, SeriesType)
+
+    if not isinstance(deep, (types.Omitted, types.Boolean)) and not deep:
+        ty_checker.raise_exc(self.data, 'boolean', 'deep')
+
+    if isinstance(self.index, types.NoneType):
+        def hpat_pandas_series_copy_impl(self, deep=True):
+            if deep:
+                return pandas.Series(data=self._data.copy(), name=self._name)
+            else:
+                return pandas.Series(data=self._data, name=self._name)
+        return hpat_pandas_series_copy_impl
+    else:
+        def hpat_pandas_series_copy_impl(self, deep=True):
+            if deep:
+                return pandas.Series(data=self._data.copy(), index=self._index.copy(), name=self._name)
+            else:
+                return pandas.Series(data=self._data, index=self._index.copy(), name=self._name)
```

Comment:
Why do we copy index here with deep=False? Is it only because we have no immutable indexes now?
Anyway, can we add a comment here for clarity.

Reply:
Oh, I see in the test below: "# Shallow copy of index is not supported yet". But still a comment here would be useful.

Reply:
done